_posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md

```diff
@@ -0,0 +1,71 @@
+date: 2017-11-07 00:03:30 +0200
+categories: haskell how-we-learn
+author_name: Sergii Paryzhskyi
+author_url : /author/sergii_paryzhskyi
+author_avatar: sergii_paryzhskyi
+show_avatar : true
+read_time : 7
+show_related_posts: false
+square_related: recommend-wolf
+feature_image: posts/2017-10-17-first-haskell-meeting/haskell-poster.jpg
+---
+
+This post is a quick update and the latest news about our Haskell learning group. Even though there were no blog posts to this topic recently, we are still having weekly meetings and mostly stick to the plan. That of course implies preparation time, spend between our sessions. There is no rush in our learning process and we take time to discuss interesting topics a bit longer, especially when all participants are involved. That being said, we might have a plan to go through one chapter per session but it also fine if it is less than that.
+
+One thing we've noticed and that was to expect, is that the difficulty of a content is slowly increasing. We are spending more time reading chapters. Sometimes the same material has to be read multiple times in order to fully understand what is it about.
+
+It helps a lot that the most of the concepts are familiar to us through other languages or libraries we are using. Let's say JavaScript doesn't support currying natively the way Haskell does it, but there are libraries which introduce these concepts. We are very often using [RamdaJS][ramdajs] in our projects, so such things as function composition, currying, partial application and so on were not something totally new to us. They were still had to be "translated" and demonstrated in a Haskell way but we certainly didn't need much time to get the idea.
+
+Similar situation was with typing system and type definitions. As many people from the group doing i.e. TypeScript or Scala, there were not that many surprises for us. Nevertheless it's pretty impressive what can be achieved with type constraints in Haskell.
+
+[ramdajs]: http://ramdajs.com
+
+## Types
+
+We went through many topics in the book and since the last update we've covered chapter four, five and discussed a half of the chapter six which covers Typeclasses. The main topics were:
+* Basic DataTypes
+* Types
+* Typeclasses
+
+We learned important basics that helps a lot when you read and try to understand what certain code does. For instance understanding how to read type signature, so something like this starting to make sense:
+
+`(Num a, Num b) => a -> b -> b`
+
+In this case there are two constraints, one for each parameter. They must be of a typeclass Num. You might already have some idea what function might do, based on its type signature.
+We had a look on different examples of how type signatures limit a scope of what body of the function might be. Even more, some  type signatures of function tells us exactly what it does because its the only possible implementation of the function. The simplest example of that will be an identity function:
```

Comment:
`[... ]some  type signatures of function [...]` contains double space between `some` and `type` :)

Reply:
I have no idea how you noticed that :) thx

Reply:
hah! no problemo :) I just read this carefully and with interest! :)